statsmodels/graphics/functional.py

```diff
@@ -1,14 +1,315 @@
 """Module for functional boxplots."""
 
+from statsmodels.multivariate.pca import PCA
+from statsmodels.nonparametric.kernel_density import KDEMultivariate
 from statsmodels.compat.python import combinations, range
 import numpy as np
-from scipy import stats
 from scipy.misc import factorial
+import matplotlib.pyplot as plt
 
 from . import utils
 
 
-__all__ = ['fboxplot', 'rainbowplot', 'banddepth']
+__all__ = ['hdrboxplot', 'fboxplot', 'rainbowplot', 'banddepth']
+
+
+def kernel_smoothing_(data, optimize=False):
+    """Create gaussian kernel.
+
+    Parameters
+    ----------
+    data : sequence of ndarrays or 2-D ndarray
+        The vectors of functions to create a functional boxplot from.  If a
+        sequence of 1-D arrays, these should all be the same size.
+        The first axis is the function index, the second axis the one along
+        which the function is defined.  So ``data[0, :]`` is the first
+        functional curve.
+    optimize : bool, optional
+        Use `normal_reference` or `cv_ml`. Default is False.
+
+    Returns
+    -------
+    kde : KDEMultivariate instance
+
+    """
+    _, dim = data.shape
+
+    if optimize:
+        kde = KDEMultivariate(data, bw='cv_ml', var_type='c' * dim)
+    else:
+        kde = KDEMultivariate(data, bw='normal_reference', var_type='c' * dim)
+
+    return kde
+
+
+def inverse_transform_(pca, data):
+    """Inverse transform on PCA.
+
+    Use PCA's `project` method by temporary replacing its factors with
+    `data`.
+
+    Parameters
+    ----------
+    pca : statsmodels Principal Component Analysis instance
+        The PCA object to use.
+    data : sequence of ndarrays or 2-D ndarray
+        The vectors of functions to create a functional boxplot from.  If a
+        sequence of 1-D arrays, these should all be the same size.
+        The first axis is the function index, the second axis the one along
+        which the function is defined.  So ``data[0, :]`` is the first
+        functional curve.
+
+    Returns
+    -------
+    projection : array
+        nobs by nvar array of the projection onto ncomp factors
+
+    """
+    factors = pca.factors
+    pca.factors = data.reshape(-1, factors.shape[1])
+    projection = pca.project()
+    pca.factors = factors
+    return projection
+
+
+def hdrboxplot(data, ncomp=2, alpha=[], threshold=0.95, optimize=False,
+               n_contours=50, xdata=None, labels=None, ax=None):
+    """Plot High Density Region boxplot.
+
+    1. Compute a multivariate kernel density estimation,
+    2. Compute contour lines for percentiles 90%, 50% and `alpha`%,
+    3. Plot the bivariate plot,
+    4. Compute mediane curve along with percentiles and outliers curves.
+
+    Behind the scene, the dataset is represented as a matrix. Each line
```

Comment:
long explanation should go in the notes section. This should have just be a short explanation , so we get faster to Parameter section which we/users need to look at more often